eventEmitter.js

```diff
@@ -0,0 +1,42 @@
+class EventEmitter {
+   listeners = {};
+     
+   on(eventName, fn) {
+      if(!this.listeners[eventName]) {
+         this.listeners[eventName] = [];
+      } 
+      this.listeners[eventName].push(fn) 
```

Comment:
It's preferable to use immutability to mutation. Instead of pushing to the array and mutating it, try to create a new array with the new listener at the end of it using the spread operator.